pkg/storage/couchdb/couchdbstore.go

```diff
@@ -479,6 +519,49 @@ func (c *CouchDBStore) getRawDoc(k string) (map[string]interface{}, error) {
 	return rawDoc, nil
 }
 
+func (c *CouchDBStore) getRawDocs(keys []string) ([]map[string]interface{}, error) {
+	rawDocs := make([]map[string]interface{}, len(keys))
+
+	bulkGetReferences := make([]kivik.BulkGetReference, len(keys))
+
+	for i, key := range keys {
+		bulkGetReferences[i].ID = key
+	}
+
+	rows, err := c.db.BulkGet(context.Background(), bulkGetReferences)
+	if err != nil {
+		return nil, err
+	}
+
+	ok := rows.Next()
+
+	if !ok {
+		return nil, errors.New("bulk get from CouchDB was unexpectedly empty")
+	}
+
+	for i := 0; i < len(rawDocs); i++ {
+		err := rows.ScanDoc(&rawDocs[i])
+		if err != nil && !strings.Contains(err.Error(), bulkGetDocNotFoundErrMsgFromKivik) {
+			return nil, fmt.Errorf(failureWhileScanningResultRowsDoc, err)
+		}
+
+		ok := rows.Next()
+
+		// ok is expected to be false on the last doc.
+		if i < len(rawDocs)-1 {
+			if !ok {
+				return nil, errors.New("got fewer docs from CouchDB than expected")
+			}
+		} else {
+			if ok {
```

Comment:
How can this happen?

Reply:
Technically this should be impossible unless CouchDB or the Kivik library has some sort of bug. Consider this an extra check just to make sure, and if something does go wrong I'll get a more specific error message.

(the "bulk get from CouchDB was unexpectedly empty" check above the loop is in the same category of "should be impossible)